src/core/scroll/scroll.service.js

```diff
@@ -0,0 +1,188 @@
+import { AppError } from '../infrastructure/error';
+import { jobLine } from '../services/job.line';
+import { PathService } from '../path/path.service';
+
+export class ScrollService {
+    constructor(model, table, bag, view) {
+        this.model = model;
+        this.table = table;
+        this.bag = bag;
+        this.view = view;
+        this.interval = null;
+        this.rect = null;
+        this.body = null;
+        this.job = jobLine(0);
+        this.startCell = null;
+        this.mousePosition = null; 
+
+        const pathFinder = new PathService(bag.body);
+        model.scrollChanged.watch(e => {
+            if (this.interval) {              
+                const path = this.getPath(this.mousePosition);
+                const td = pathFinder.cell(path);
+
+                this.navigate(td);
+                this.view.selection.selectRange(this.startCell, td, 'body');
+            }
+		});
+    }
+
+    canScroll(e) {
+        const table = this.rect;
+        const offset = this.offset;
+
+        return !(e.clientY < (table.bottom - offset) &&
+            e.clientY > (table.top + offset) &&
+            e.clientX > (table.left + offset) &&
+            e.clientX < (table.right - offset))
+    }
+
+    scroll(e, startCell) {
+        this.mousePosition = e;
+        this.startCell = startCell; 
+
+        if (!this.body) {
+            this.invalidate();
+        }
+
+        if (this.interval) {
+            if (!this.canScroll(e)) {
+                this.clearInterval();
+                return;
+            }
+        }
+
+        const direction = this.onEdgeOf(e);
+        if (direction && !this.interval) {
+            this.job(() => {
+                const interval = this.doScroll(direction);
+                this.interval = interval();
+            })
+        }
+
+    }
+
+    doScroll(direction) {
+        const { scroll } = this.model;
+        const scrollState = scroll();
+        const { velocity }  = scrollState;
+        const scrolledToEnd = () => this.isScrolledToEnd(direction);
+
+        return () => setInterval(() => {
```

Comment:
try to use job

Reply:
setInterval is located in function, which is already wrapped in Job. 
![image](https://user-images.githubusercontent.com/13670003/44781553-60593e80-ab8d-11e8-9faf-8b6bdcfd709f.png)

Reply:
this is to tricky, we need to make own interval like a jobLine, or emulate it with job